tests/test_hawkdovevar.py

```diff
@@ -0,0 +1,150 @@
+import statistics
+from unittest.mock import patch, Mock
+
+import pytest
+
+from simulatingrisk.hawkdovevar.model import (
+    HawkDoveVariableRiskModel,
+    HawkDoveVariableRiskAgent,
+)
+
+
+def test_init():
+    model = HawkDoveVariableRiskModel(5)
+    # defaults
+    assert model.risk_adjustment is None
+    assert model.hawk_odds == 0.5
+    assert model.include_diagonals is True
+    # unused but should be set to default
+    assert model.adjust_round_n == 10
+
+    # init with risk adjustment
+    model = HawkDoveVariableRiskModel(
+        5,
+        include_diagonals=False,
+        hawk_odds=0.2,
+        risk_adjustment="adopt",
+        adjust_every=5,
+    )
+
+    assert model.risk_adjustment == "adopt"
+    assert model.adjust_round_n == 5
+    assert model.hawk_odds == 0.2
+    assert model.include_diagonals is False
+
+    # handle string none for solara app parameters
+    model = HawkDoveVariableRiskModel(5, risk_adjustment="none")
+    assert model.risk_adjustment is None
+
+    # complain about invalid adjustment type
+    with pytest.raises(ValueError, match="Unsupported risk adjustment 'bogus'"):
+        HawkDoveVariableRiskModel(3, risk_adjustment="bogus")
+
+
+def test_init_variable_risk_level():
+    model = HawkDoveVariableRiskModel(
+        5,
+        include_diagonals=True,
+    )
+    # when risk level is variable/random, agents should have different risk levels
+    risk_levels = set([agent.risk_level for agent in model.schedule.agents])
+    assert len(risk_levels) > 1
+
+
+adjustment_testdata = [
+    # init parameters, expected adjustment round
+    ({"risk_adjustment": None}, None),
+    ({"risk_adjustment": "adopt"}, 10),
+    ({"risk_adjustment": "average"}, 10),
+    ({"risk_adjustment": "average", "adjust_every": 3}, 3),
+]
+
+
+@pytest.mark.parametrize("params,expect_adjust_step", adjustment_testdata)
```

Comment:
This is nice, so it runs the test 4 times according to the number of tuples in the `adjustment_testdata` list?

Reply:
yeah, isn't that cool? very useful sometimes